forecasts.py

```diff
@@ -0,0 +1,157 @@
+""" This module contain logic to retrieve predicted weather data.
+"""
+from os import getenv
+from datetime import datetime, timedelta
+import asyncio
+import json
+import logging
+from io import StringIO
+from typing import List
+import pandas
+from aiohttp import ClientSession
+from schemas import WeatherForecast, WeatherStation, WeatherForecastValues
+
+LOGGER = logging.getLogger(__name__)
+
+
+class StationNotFoundException(Exception):
+    """ Custom exception for when a station cannot be found """
+
+
+def fetch_station_by_code(station_code: int) -> WeatherStation:
+    """ Return a station matching the specified code """
+    # NOTE: This will change to use the WildeFire API.
+    with open('data/weather_stations.json') as file_pointer:
+        stations = json.load(file_pointer)
+        for station in stations['weather_stations']:
+            if int(station['code']) == station_code:
+                return WeatherStation(**station)
+        raise StationNotFoundException()
+
+
+async def fetch_forecast(session: ClientSession, station_code: int) -> WeatherForecast:
+    """ Return the forecast for a weather station from spotwx.
+    """
+    # We need the station coordinates, spotwx doesn't know about station id's.
+    station = fetch_station_by_code(station_code)
+
+    url = '{base_url}?key={key}&model={model}&lat={lat}&lon={long}'.format(
+        lat=station.lat, long=station.long, model='gdps', key=getenv('SPOTWX_API_KEY'),
+        base_url=getenv('SPOTWX_BASE_URI'))
+
+    # Key value map from spotwx to out structure.
+    key_map = {
```

Comment:
Feels like this variable can live somewhere else other than inside here

Reply:
Yep - I'll move - I also don't like that we actually send on a bunch of values that are just ignore.

Reply:
Yeah I suppose we can comment out some properties that are not being used. If you are going this route let me know.